From 66e4f4b05f744f2d37caae98749384029b118922 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Thu, 11 May 2023 14:44:51 +0800
Subject: [PATCH] [riscv]Enable caged heap

Change-Id: I6aacb3447855b1910a291b37737a28443239cb4e
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4521076
Reviewed-by: Jakob Kummerow <jkummerow@chromium.org>
Commit-Queue: Yahan Lu <yahan@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#87587}
---
 BUILD.gn | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/BUILD.gn b/BUILD.gn
index 0d78268041..9471e173b9 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -303,7 +303,7 @@ declare_args() {
   # Enable heap reservation of size 4GB. Only possible for 64bit archs.
   cppgc_enable_caged_heap =
       v8_current_cpu == "x64" || v8_current_cpu == "arm64" ||
-      v8_current_cpu == "loong64"
+      v8_current_cpu == "loong64" || v8_current_cpu == "riscv64"
 
   # Enables additional heap verification phases and checks.
   cppgc_enable_verify_heap = ""
@@ -658,7 +658,8 @@ assert(!v8_enable_unconditional_write_barriers || !v8_disable_write_barriers,
        "Write barriers can't be both enabled and disabled")
 
 assert(!cppgc_enable_caged_heap || v8_current_cpu == "x64" ||
-           v8_current_cpu == "arm64" || v8_current_cpu == "loong64",
+           v8_current_cpu == "arm64" || v8_current_cpu == "loong64" ||
+           v8_current_cpu == "riscv64",
        "CppGC caged heap requires 64bit platforms")
 
 assert(!cppgc_enable_young_generation || cppgc_enable_caged_heap,
-- 
2.35.1

